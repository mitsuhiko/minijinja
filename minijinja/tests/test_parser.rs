#![cfg(feature = "unstable_machinery")]
use minijinja::machinery::parse;

#[test]
fn test_parser() {
    insta::glob!("parser-inputs/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let filename = path.file_name().unwrap().to_str().unwrap();
        let ast = parse(&contents, filename, Default::default(), Default::default());
        insta::with_settings!({
            description => contents.trim_end(),
            omit_expression => true,
        }, {
            insta::assert_debug_snapshot!(&ast);
        });
    });
}
