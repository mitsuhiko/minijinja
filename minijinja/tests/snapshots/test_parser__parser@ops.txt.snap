---
source: minijinja/tests/test_parser.rs
description: "{{ foo and bar or baz == true }}"
input_file: minijinja/tests/parser-inputs/ops.txt
---
Ok(
    Template {
        children: [
            EmitExpr {
                expr: BinOp {
                    op: ScOr,
                    left: BinOp {
                        op: ScAnd,
                        left: Var {
                            id: "foo",
                        } @ 1:3-1:6,
                        right: Var {
                            id: "bar",
                        } @ 1:11-1:14,
                    } @ 1:3-1:14,
                    right: BinOp {
                        op: Eq,
                        left: Var {
                            id: "baz",
                        } @ 1:18-1:21,
                        right: Const {
                            value: true,
                        } @ 1:25-1:29,
                    } @ 1:15-1:29,
                } @ 1:3-1:29,
            } @ 1:0-1:29,
        ],
    } @ 0:0-1:32,
)
